Information gain changes for Attribution Scope

feifeiji89 opened the pull request:
The current attribution logic in the Attribution Reporting API may not be ideal for use-cases where an ad-tech needs more fine grain control over the attribution granularity (i.e. campaign, product, conversion ID, etc.) before a source is chosen for attribution. Currently available features such as top-level filters are not fully sufficient for this use-case because they happen after a source has been selected (i.e. after attribution). We can optionally support this use-case by allowing callers of the API to specify predefined attribution scopes that will be considered for filtering before attributing a source, in order to more efficiently extract utility out of the API.

This PR covers the information gain computation logic.

linnan-github committed:
Add non-normative note for event-level reports deletion for unexpired destination limit (#1351)

Co-authored-by: Andrew Paseltiner <apaseltiner@google.com>

apasel422 committed:
Rename summary window operator to summary operator (#1346)

Fixes #1343

giladbarkan-github committed:
Allow max attribution rate limit when replacing an event report (#1326)

Move attribution rate-limit and event storage limit checks

linnan-github committed:
Fix attribution_scopes explainer for size limits on triggers (#1360)

giladbarkan-github commented on index.bs:
```diff
@@ -2392,6 +2392,11 @@ To <dfn>obtain a randomized source response pick rate</dfn> given a [=randomized
 1. Let |numPossibleValues| be the [=set/size=] of |possibleValues|.
 1. Return |numPossibleValues| / (|numPossibleValues| - 1 + e<sup>|epsilon|</sup>).
 
+To <dfn>obtain pick rate for a source with attribution scope</dfn> given a positive integer |numTriggerStates|, a positive integer |numEventStates|, a non-negative 32-bit integer |numUsedScopes|, and a double |epsilon|:
```
This method name seems a bit confusing since we always use it but generally we define two types of sources, with and without attribution scope.

giladbarkan-github reviewed (approved):
I reviewed the spec, not the JavaScript sections, which seems in line.

csharrison committed:
Create 2024-06-24-minutes.md

feifeiji89 committed:
Parsing changes for Pre-Attribution Filtering: Attribution Scope (#1352)

Co-authored-by: Andrew Paseltiner <apaseltiner@google.com>

apasel422 committed:
Clean up attribution-scope parsing (#1363)

- Split scopes-parsing for source and trigger into separate paths
- Emit an error when fields cannot be fully validated due to invalid attribution_scope_limit
- Remove unnecessary lambdas
- Make parsing functions use conventional parameter ordering

feifeiji89 committed:
Information gain computation logic for Attribution Scope.
